tools/distpartitioning/multi_dev_init.py

```diff
@@ -0,0 +1,513 @@
+import os
+import sys
+import constants
+import numpy as np
+import math
+import torch
+import torch.distributed as dist
+import torch.multiprocessing as mp
+import dgl
+
+from timeit import default_timer as timer
+from datetime import timedelta
+from dataset_utils import get_dataset
+from utils import read_partitions_file, read_json, get_node_types, \
+                    augment_node_data, augment_edge_data, get_ntypes_map, \
+                    write_dgl_objects, write_metadata_json
+from gloo_wrapper import alltoall_cpu_object_lst, alltoallv_cpu, \
+                    alltoall_cpu, allgather_sizes, gather_metadata_json
+from globalids import assign_shuffle_global_nids_nodes, \
+                    assign_shuffle_global_nids_edges, get_shuffle_global_nids_edges
+from convert_partition import create_dgl_object, create_metadata_json, validateDGLObjects
+
+def exchange_node_data(rank, world_size, node_data):
+    """
+    Exchange node_data among the processes in the world
+    Prepare the list of slices targeting each of the process and
+    trigger alltoallv_cpu for the message exchange.
+
+    Parameters:
+    -----------
+    rank : int
+        rank of the current process
+    world_size : int
+        total no. of participating processes
+    node_data : dictionary
+        nodes data dictionary with keys as column names and values as
```

Comment:
Can you provide more information of node_data here? For example what do OWNER_PROCESS, GLOBAL_NID, etc mean.